src/summary_eval.py

```diff
@@ -92,31 +95,36 @@ def score_wording(self) -> None:
             self.input_text, truncation=True, max_length=4096
         )[0]["score"]
 
-    def extract_keyphrases(self) -> None:
+    def suggest_keyphrases(self) -> None:
         """Return keyphrases that were included in the summary and suggests
         keyphrases that were not included.
         """
-        included_keyphrases = set()
+        included_keyphrases = list()
         suggested_keyphrases = list()
+        weights = list()
 
-        summary_lemmas = {t.lemma_ for t in self.summary if not t.is_stop}
+        summary_lemmas = " ".join(
+            [t.lemma_.lower() for t in self.summary if not t.is_stop]
+        )
 
-        for keyphrase in self.keyphrases:
-            key_lemmas = {t.lemma_ for t in keyphrase if not t.is_stop}
-            keyphrase_included = not summary_lemmas.isdisjoint(key_lemmas)
-            if keyphrase_included:
-                included_keyphrases.add(keyphrase.text)
-            else:
-                suggested_keyphrases.append(keyphrase.text)
+        for chunk in self.chunks.values():
+            for keyphrase in chunk["keyphrases"]:
+                keyphrase_lemmas = [t.lemma_ for t in keyphrase if not t.is_stop]
+                keyphrase_included = re.search(
+                    re.escape(r" ".join(keyphrase_lemmas)),
+                    summary_lemmas,
+                    re.IGNORECASE,
+                )
+                if keyphrase_included:
+                    included_keyphrases.append(keyphrase.text)
+                elif keyphrase.text not in suggested_keyphrases:

```

Comment:
I feel like keyphrases that appear across multiple chunks should be given extra weight. One way could be to eliminate the 'in suggested_keyphrases' check, so keyphrases that appear two times will be given double the chance of being suggested:

```suggestion
else:
```

Reply:
Clever idea, but I think this could result in suggesting duplicate keyphrases. It is probably better to += the weight if keyphrase in suggested_keyphrases.